dbt/adapters/bigquery.py

```diff
@@ -111,6 +118,9 @@ def open_connection(cls, connection):
         result = connection.copy()
         credentials = connection.get('credentials', {})
 
+        if 'timeout_seconds' in credentials:
+            cls.QUERY_TIMEOUT = credentials['timeout_seconds']
```

Comment:
don't mutate global state. a better way:

- make `poll_until_job_completes` take a timeout as an argument
- then call it with `poll_until_job_completes(job, timeout=connection.get('credentials', {}).get('timeout_seconds'), cls.DEFAULT_QUERY_TIMEOUT)`

Reply:
(and remove this logic altogether)